build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```diff
@@ -62,14 +62,18 @@ class Capture {
     boolean ptsSupported(Test t, Set<String> engines) {
         if (!engines.isEmpty() && engines.stream().allMatch { e -> supportedEngines.containsKey(e) }) {
             // If cucumber is used without companion it's not supported, otherwise it is.
-            return !(cucumberUsed(t, engines) && !t.project.plugins.hasPlugin('com.gradle.cucumber.companion'))
+            return !(cucumberUsed(t) && !t.project.plugins.hasPlugin('com.gradle.cucumber.companion'))
         } else {
             return false
         }
     }
 
-    boolean cucumberUsed(Test t, Set<String> engines) {
-        return t.classpath.any { f -> f.name.contains('cucumber')} || engines.any { it.contains('cucumber')}
+    boolean cucumberUsed(Test t) {
+        return t.project.configurations.stream().filter { it.canBeResolved }.any {
```

Comment:
Is the `testImplementation` always a resolvable dependency? I'm wondering if there are cases where this check would just return `false` or `true` incorrectly if it isn't resolvable.

Reply:
I don't think `testImplementation` is resolvable in the same way that `implementation` is not resolvable. They just declare a group of dependencies. Can you please check?

This is a useful blog post that someone in our slack linked to that covers this topic.
https://dev.to/autonomousapps/configuration-roles-and-the-blogging-industrial-complex-21mn

Reply:
Is this method iterating through the dependencies of `testImplementation`? y/n? You can debug this if needed with some printlines (or tagging).

Reply:
No, for a simple cucumber test project it's checking 
```
annotationProcessor
compileClasspath
cucumberRuntime
runtimeClasspath
testAnnotationProcessor
testCompileClasspath
testRuntimeClasspath
```

Reply:
I see, I think we only need to check the `test` configurations. For example, if the `compileClasspath` contains cucumber then it isn't necessarily being used in a test (perhaps we're building a cucumber based plugin or framework).

Reply:
Ok since there isn't a good way to detect only the test configuration and we assume that this script will be used with a solutions engineer present this is okay.